core/validation/specific_validations/towns_fund_round_four.py

```diff
@@ -374,39 +373,46 @@ def validate_funding_spent(workbook: dict[str, pd.DataFrame]) -> list["TownsFund
             ]
     else:
         # check funding against individual project funding allocated for Towns Deal submissions
-        return [
-            TownsFundRoundFourValidationFailure(
-                sheet="Funding",
-                section=f"Project Funding Profiles - Project {get_project_number_by_id(project_id, project_ids)}",
-                column="Grand Total",
-                message=msgs.OVERSPEND_PROJECT,
-                # first project begins after 17 and each project is seperated by 28 cells
-                row_index=17 + 28 * get_project_number_by_id(project_id, project_ids),
-            )
-            for project_id in project_ids
-            if funding_spent[project_id] > DefaultConfig.TF_FUNDING_ALLOCATED[project_id]
-        ]
-
-    return funding_spent_failures
+        funding_spent_failures = []
+        for expense_type in ["CDEL", "RDEL"]:
+            for idx in project_ids:
+                project_number = get_project_number_by_id(idx, project_ids)
+                if funding_spent[expense_type][idx] > DefaultConfig.TF_FUNDING_ALLOCATED[expense_type][idx]:
+                    funding_spent_failures.append(
+                        TownsFundRoundFourValidationFailure(
+                            sheet="Funding",
+                            section=f"Project Funding Profiles - Project {project_number}",
+                            column="Grand Total",
+                            message=msgs.OVERSPEND.format(spend_type="CDEL" if expense_type == "CDEL" else "RDEL"),
```

Comment:
Any reason why we dont do
`message=msgs.OVERSPEND.format(spend_type=expense_type)`
instead of 
`message=msgs.OVERSPEND.format(spend_type="CDEL" if expense_type == "CDEL" else "RDEL")`

could we also align the variable names to either `spend_type` or `expense_type`